CIPs/cip-granda_mento.md

```diff
@@ -0,0 +1,186 @@
+
+## Motivation
+
+There are no existing avenues that are able to satisfy high volume stable token exchanges on the order of millions. Users looking to make larger exchanges involving stable tokens are faced with 3 options:
+1. Exchange `CELO <-> stable token` via Mento slowly over a longer period of time.
+2. Arrange one or many `XXX <-> stable token` OTC trades.
+3. Create a large limit order of `fiat <-> stable token` on a centralized exchange in hope it gets filled over time.
+
+The existing implementation of Mento is suitable for facilitating low volume exchanges, e.g. up to thousands of stable tokens at a time, with minimal slippage. For cUSD, slippage of over 2% starts to occur on both Mento and on centralized exchanges at trade sizes of ~$50k+. OTC trading satisfies the needs of medium volume exchanges, e.g. ~$100k, but similarly takes a hit of around 2-3%. For lower total supply stable tokens with lower liquidity, larger OTC trades or centralized exchange orders may not be possible.
+
+Increasing Mento bucket sizes permanently to allow larger exchanges via Mento will reduce slippage, but it puts a more significant portion of the Reserve at risk.
+
+Building a process to exchange large (~$1 million+) amounts of CELO for stable tokens will enable large entities to make significant block purchases of stable tokens at scale. While minting of the stable token is expected to be the main use case, reversibility (ie exchanging `stable token -> CELO`) is desirable to instill higher confidence when exchanging large quantities.
+
+While Granda Mento will help facilitate `CELO <-> stable token` exchanges, this is ultimately meant as a mechanism to help purchasers with fiat (e.g. USD) who are seeking to own Celo's corresponding stable token (e.g. cUSD). Purchasers would come to an agreement with a broker (e.g. the Celo Foundation) who owns an existing amount of CELO. The broker would be the one to make the `CELO -> cUSD` exchange on-chain.
+
+The proposed implementation is being considered in the medium-term for large scale exchanges until Mento is possibly reworked to support more efficient large-scale minting on a sustained basis.
+
+## Specification
+
+At a high level, the design involves:
+1. Anyone can submit an exchange proposal.
+   * The amount of stable tokens being bought/sold in the exchange is required to be within a governable range.
+   * The assets being sold (ie either the stable token or CELO) in the exchange are deposited.
+   * The current oracle price for the exchange is recorded.
+2. The proposed exchange must be approved by a multisig that has previously been authorized by Governance. At this point, the exchange still cannot be executed.
+3. A forced waiting period of X days must elapse before the exchange can be executed. During this time, Governance can choose to veto the exchange, refunding the exchange proposer.
+4. After the waiting period, the exchange is executed according to the oracle price recorded in step 1.
```

Comment:
Shall it include a fee? I see no reason why this wouldn't include a fee of 0.5%, because it's not paying any spread and indeed bigger trades would be more favorable than small ones.

There could be an argument that the fee should be lower than 0.5%, since the exchanger is bearing risk in case of a big price move against their favor, but the whole exchange can be vetoed in case the price moves on their favor.

Reply:
That's the purpose of `FixidityLib.Fraction public spread;` further down -- I'll make this clearer here

Reply:
Oh, missed that